util/hot_backup_test.cc

```diff
@@ -0,0 +1,600 @@
+// -------------------------------------------------------------------
+//
+// hot_backup_test.cc
+//
+// Copyright (c) 2016 Basho Technologies, Inc. All Rights Reserved.
+//
+// This file is provided to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file
+// except in compliance with the License.  You may obtain
+// a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+// -------------------------------------------------------------------
+
+#include <fcntl.h>
+#include <libgen.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "leveldb/db.h"
+#include "util/testharness.h"
+#include "util/testutil.h"
+
+#include "db/dbformat.h"
+#include "db/db_impl.h"
+#include "db/filename.h"
+#include "util/hot_backup.h"
+
+
+/**
+ * Execution routine
+ */
+int main(int argc, char** argv)
+{
+    int ret_val;
+
+    ret_val=leveldb::test::RunAllTests();
+
+    return(ret_val);
+}   // main
+
+
+namespace leveldb {
+
+
+/**
+ * Wrapper class for tests.  Holds working variables
+ * and helper functions.
+ */
+class HotBackupTester : public HotBackup
+{
+public:
+    std::string m_DBName;
+    std::string m_Trigger;
+
+    HotBackupTester()
+    {
+        m_DBName = test::TmpDir() + "/hot_backup";
+        m_Trigger = test::TmpDir() + "/trigger";
+        gHotBackup=this;
+    };
+
+    virtual ~HotBackupTester()
+    {
+        ResetHotBackupObject();
+    };
+
+    virtual const char * GetTriggerPath() {return(m_Trigger.c_str());};
+
+    void ClearAllBackups(
+        Options & InOptions)
+    {
+        int loop;
+        Options options;
+
+        // clear old if exists
+        for (loop=0; loop<=config::kNumBackups; ++loop)
+        {
+            options=InOptions;
+
+            SetBackupPaths(options, loop);
+            DestroyDB("", options);
+        }   // if
+
+        options.tiered_slow_level=4;
+        options.tiered_fast_prefix=m_DBName + "/fast";
+        options.tiered_slow_prefix=m_DBName + "/slow";
+        DestroyDB("", options);
+    }   // ClearAllBackups
+
+};  // class HotBackupTester
+
+
+/**
+ * Initial code used the existance of /etc/riak/hot_backup file
+ *  as flag to start a backup.
+ */
+TEST(HotBackupTester, FileTriggerTest)
+{
+    char * dup_path, *path;
+    int ret_val;
+    bool ret_flag;
+    uint64_t perf_before, perf_after;
+
+    perf_before=gPerfCounters->Value(ePerfSyslogWrite);
+
+    // cleanup anything existing, likely fails
+    ///  hmm, should there be a way to move this trigger
```

Comment:
It looks like the HotBackupTester class sets a trigger file path in a temp directory, so I'm guessing this is an old comment (that can be removed).